### PR TITLE
doc: document cue fmt -s in the package coding style

### LINKED ISSUE #2434: `cue fmt -s` aka `--simplify` in CI linter is confusing for users making PRs
`cue fmt -s` aka `--simplify` in linter is confusing for users making PRs with valid CUE code that doesn't pass.

https://github.com/dagger/dagger/pull/2433#issuecomment-1124550442

If this is important to us, it should be be documented somewhere like https://docs.dagger.io/1226/coding-style and users need better feedback/error messages in GitHub.

cc @samalba

### PR DESCRIPTION
**Pull request by samalba**
Fixes #2434

### CODE REVIEW

File: docs/guides/1226-coding-style.md [hunk 1]
```diff
@@ -381,3 +381,13 @@ client: _
...
+
+## Format all the CUE code
+
+It's important to lint all the CUE code so it stays consistent across all packages.
+
+Thanksfully, CUE makes it very easy. [First, make sure you have the CUE binary installed on your machine](https://cuelang.org/docs/install/). Then, simply run the following command from the package directory:
+
+```shell
+cue fmt -s
+```
```
Comment (TomChv) [inline docs/guides/1226-coding-style.md:393]: They can also run `make cuefmt` from the root of dagger repository, should we documente that one?
They can also run `dagger do lint cue` to only find error.

**Review by TomChv — COMMENTED** [review]
That's a good idea, I left a question

File: docs/guides/1226-coding-style.md (hunk 1 above)
Comment (shykes) [reply docs/guides/1226-coding-style.md:393]: * `dagger do lint cue` 👍
* `make cuefmt` 👎 we should remove all dependencies on our Makefile in preparation for removing it altogether

**Review by shykes — APPROVED** [review]

File: docs/guides/1226-coding-style.md (hunk 1 above)
Comment (samalba) [reply docs/guides/1226-coding-style.md:393]: agreed with @shykes for pushing dagger when it's about code on the dagger repo. The reason why I suggest `cue fmt -s` here is to make it work outside of our repo, I assume you may want to write new packages on any repo, so I'd say our internal tooling is out of scope here.

Comment (shykes) [reply docs/guides/1226-coding-style.md:393]: Yes I agree `cue fmt -s` 👍

**Pull request merged**